posthog/temporal/batch_exports/temporary_file.py

```diff
@@ -0,0 +1,523 @@
+"""This module contains a temporary file to stage data in batch exports."""
+import abc
+import collections.abc
+import contextlib
+import csv
+import datetime as dt
+import gzip
+import tempfile
+import typing
+
+import brotli
+import orjson
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+
+def json_dumps_bytes(d) -> bytes:
+    return orjson.dumps(d, default=str)
+
+
+class BatchExportTemporaryFile:
```

Comment:
This class is copied over from `batch_expots.py`. No changes except that finishing brotli compressor is now in a separate function. Destinations except S3 will keep on using this until we refactor them and then we can delete all the `write+` methods from here.